src/excmds.ts

```diff
@@ -883,9 +883,23 @@ export async function restart() {
  * @param filename The name the file should be saved as.
  */
 //#content
-export async function saveas(...filename: string[]) {
-    if (filename.length > 0) {
-        return Messaging.message("download_background", "downloadUrlAs", window.location.href, filename.join(" "))
+export async function saveas(...args: string[]) {
+    if (args.length > 0) {
+        let overwrite = false
+        let cleanup = false
+        const uniq_args_set = new Set(args)
+
+        if (uniq_args_set.has("--overwrite")) {
+            overwrite = true
+            uniq_args_set.delete("--overwrite")
+        }
+
+        if (uniq_args_set.has("--cleanup")) {
+            cleanup = true
+            uniq_args_set.delete("--cleanup")
+        }
```

Comment:
These need `nativegate( [whatever version of the Nim messenger gets them, probably 0.3?] )` : )

Reply:
@bovine3dom Added in https://github.com/gsbabil/tridactyl_native_messenger/commit/043488b3cff5943678ed62bbbddda7ca3bf90ad6